modules/integration/tests-integration/bpmn/src/test/java/org/wso2/bps/integration/tests/bpmn/ProcessInstanceTestCase.java

```diff
@@ -0,0 +1,70 @@
+import org.wso2.bps.integration.common.utils.BPSMasterTest;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+
+import java.io.File;
+
+public class ProcessInstanceTestCase extends BPSMasterTest {
+
+    @Test(groups = {"wso2.bps.test.processInstance"}, description = "Process Instance Test", priority = 2, singleThreaded = true)
+    public void ProcessInstanceTests() throws Exception {
+        init();
+        ActivitiRestClient tester = new ActivitiRestClient(bpsServer.getInstance().getPorts().get("http"), bpsServer.getInstance().getHosts().get("default"));
+
+        //deploying Package
+        String filePath = FrameworkPathUtil.getSystemResourceLocation() + File.separator
+                          + BPMNTestConstants.DIR_ARTIFACTS + File.separator
+                          + BPMNTestConstants.DIR_BPMN + File.separator + "HelloApprove.bar";
+        String fileName = "HelloApprove.bar";
+        String[] deploymentResponse;
+        deploymentResponse = tester.deployBPMNPackage(filePath, fileName);
+        Assert.assertTrue("Deployment Successful", deploymentResponse[0].contains(BPMNTestConstants.CREATED));
+        String[] deploymentCheckResponse = tester.getDeploymentInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Deployment Present", deploymentCheckResponse[2].contains(fileName));
+
+        //Acquiring Process Definition ID to start Process Instance
+        String[] definitionResponse = tester.findProcessDefinitionInfoById(deploymentResponse[1]);
+        Assert.assertTrue("Search Success", definitionResponse[0].contains(BPMNTestConstants.OK));
+
+        //Starting and Verifying Process Instance
+        String[] processInstanceResponse = tester.startProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Started", processInstanceResponse[0].contains(BPMNTestConstants.CREATED));
+        String searchResponse = tester.searchProcessInstanceByDefintionID(definitionResponse[1]);
+        Assert.assertTrue("Process Instance Present", searchResponse.contains(BPMNTestConstants.OK));
+
+        //Suspending the Process Instance
+        String[] suspendResponse = tester.suspendProcessInstanceById(processInstanceResponse[1]);
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[0].contains(BPMNTestConstants.OK));
+        Assert.assertTrue("Process Instance has been suspended", suspendResponse[1].contains("true"));
+
+        //Deleting a Process Instance
+        String deleteStatus = tester.deleteProcessInstanceByID(processInstanceResponse[1]);
+        Assert.assertTrue("Process Instance Removed", deleteStatus.contains(BPMNTestConstants.NO_CONTENT));
+
+        //Deleting the Deployment
+        String undeployStatus = tester.unDeployBPMNPackage(deploymentResponse[1]);
+        Assert.assertTrue("Package UnDeployed", undeployStatus.contains(BPMNTestConstants.NO_CONTENT));
+    }
```

Comment:
Whenever an operation is done, verify it with a separate call (e.g. getInstanceByID) without relying on the return value. Do this for all tests.